### PR TITLE
[ET-901] G20.09.1 Hotfix for AR Modal problems

### PR DESCRIPTION
**Pull request by sc0ttkclark**
[ET-901]

Hold on merge

Screencast: https://share.skc.dev/GGuRzY2D

[ET-901]: https://moderntribe.atlassian.net/browse/ET-901

### CODE REVIEW

File: src/Tribe/Editor/Blocks/Tickets_Item.php [hunk 1]
```diff
@@ -26,49 +26,6 @@ public function slug() {
...
-			return;
-		}
-
-		$args = array(
-			'tickets' => array_merge( $tickets, array( $ticket ) ),
-		);
-
-		// Add the rendering attributes into global context
-		$template->add_template_globals( $args );
+		return;
```
Comment (tr1b0t) [inline src/Tribe/Editor/Blocks/Tickets_Item.php:29]: :warning: Empty return statement not required here

<sub>hash: 3f81683056463ef3d89fb52bc5066f4f</sub>

Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets_Item.php:29]:
```suggestion
		// This block has no render.
```

File: src/Tribe/Editor/Blocks/Tickets_Item.php [hunk 2]
```diff
@@ -26,49 +26,6 @@ public function slug() {
...
-			return;
-		}
-
-		$args = array(
-			'tickets' => array_merge( $tickets, array( $ticket ) ),
-		);
-
-		// Add the rendering attributes into global context
-		$template->add_template_globals( $args );
+		// This block has no render.
```
Comment (cliffordp) [inline src/Tribe/Editor/Blocks/Tickets_Item.php:29]: Should return consistent with return type:
```suggestion
		// This block has no render.
		return '';
```

File: readme.txt [hunk 3]
```diff
@@ -118,6 +118,10 @@ Currently, the following add-ons are available for Event Tickets:
 
 == Changelog ==
 
+= [5.0.1] 2020-08-31 =
+
+* Fix - Prevent the RSVP provider class from interfering with the Attendee Registration modal flow in regards to a ticket's real provider. [TBD]
```
Comment (cliffordp) [inline readme.txt:123]: Doesn't feel concise. Maybe an improvement:

```suggestion
* Fix - Prevent the Attendee Registration modal from incorrectly setting RSVP as the provider class. [TBD]
```

File: readme.txt [hunk 4]
```diff
@@ -118,6 +118,10 @@ Currently, the following add-ons are available for Event Tickets:
 
 == Changelog ==
 
+= [5.0.1] 2020-08-31 =
+
+* Fix - Prevent the Attendee Registration modal from incorrectly setting RSVP as the provider class. [TBD]
```
Comment (sc0ttkclark) [inline readme.txt:123]:
```suggestion
* Fix - Prevent the Attendee Registration modal from incorrectly setting RSVP as the provider class. [ET-901]
```

**Comment by cliffordp** [comment]
@sc0ttkclark can you get non-Cest tests passing?

**Review by juanfra — APPROVED** [review]

**Comment by sc0ttkclark** [comment]
@cliffordp Both unrelated to this PR, I can get a PR that will fix some of those for G20.09, but #1686 helps towards getting things passing again

File: src/Tribe/RSVP.php [hunk 5]
```diff
@@ -1860,7 +1860,7 @@ public function get_event_for_ticket( $ticket_product ) {
 		$event_id = (int) get_post_meta( $ticket_product, $this->get_event_key(), true );
 
 		if ( 0 === $event_id ) {
-			$event_id = get_post_meta( $ticket_product, self::ATTENDEE_EVENT_KEY, true );
+			$event_id = (int) get_post_meta( $ticket_product, self::ATTENDEE_EVENT_KEY, true );
```
Comment (sc0ttkclark) [inline src/Tribe/RSVP.php:1863]: Primary issue: This was failing the next conditional below when empty

File: src/Tribe/Editor/Blocks/Tickets_Item.php [hunk 6]
```diff
@@ -26,49 +26,7 @@ public function slug() {
...
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets_Item.php:29]: Secondary issue: Tickets Item block render was polluting template arguments and making extra calls unnecessary which were then called once again in the Tickets block render.

**Pull request merged**